src/Particle/SampleStack.cpp

```diff
@@ -11,82 +11,53 @@
 
 
 #include "SampleStack.h"
-#include "Particle/MCSample.h"
 #include "Utilities/IteratorUtility.h"
 
 namespace qmcplusplus
 {
-SampleStack::SampleStack() : total_num_(0), max_samples_(10), current_sample_count_(0) {}
 
 /** allocate the SampleStack
  * @param n number of samples per rank
  * @param num_ranks number of ranks. Used to set global number of samples.
  */
-void SampleStack::setMaxSamples(int n, int num_ranks)
+void SampleStack::setMaxSamples(size_t n, size_t num_ranks)
 {
   max_samples_        = n;
   global_num_samples_ = n * num_ranks;
   //do not add anything
   if (n == 0)
     return;
   sample_vector_.reserve(n);
-  int nadd = n - sample_vector_.size();
-  while (nadd > 0)
-  {
-    sample_vector_.push_back(new MCSample(total_num_));
-    --nadd;
-  }
+  size_t nadd = n - sample_vector_.size();
+  sample_vector_.insert(sample_vector_.end(), nadd, MCSample(total_num_));
```

Comment:
What if `nadd<=0`, the old code seem robust to this.
It seem we can just do `sample_vector_.resize(nadd, MCSample(total_num_))`
instead of reserve+insert.